### PR TITLE
Vote mutations

### LINKED ISSUE #27: voteCreate GraphQL Query
Create voteCreate query to create a vote in the database

Request should be:

mutation {
  voteCreate(verseId: ID, userId: ID) {
      id
      userId
      verseId
      user  { userType }
      verse  { verseType }
    }
  }
}

Response should be:

{
    "data": {
        "voteCreate": {
             "id": "<integer>"
              "verse_id": "<integer>",
              "user_id": "<integer>",
              "user": { userType }
              "verse": { verseType }
            }
        }
    }

### PR DESCRIPTION
**Pull request by SageOfCode**
## Types of Changes

This PR closes #27 #40 

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [ ] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: spec/requests/mutations/vote_mutation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    expect(result).to have_key(:data)
+    expect(result[:data]).to be_a(Hash)
+    
+    expect(result[:data]).to have_key(:deleteVote)
+    expect(result[:data][:deleteVote]).to be_a(Hash)
+    
+    expect(result[:data][:deleteVote]).to have_key(:id)
+    expect(result[:data][:deleteVote][:id]).to be_a(String)
+  end
```
Comment (the-color-bliu) [inline spec/requests/mutations/vote_mutation_spec.rb:56]: Nice job with this test and query

File: app/graphql/mutations/delete_vote.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    argument :id, ID, required: true
+
+    type Types::VoteType
+
+    def resolve(args)
+      vote = Vote.find(args[:id])
+      vote.destroy
+      vote
+    end
+  end
```
Comment (the-color-bliu) [inline app/graphql/mutations/delete_vote.rb:14]: I'm liking what I see with the resolve method!

**Review by the-color-bliu — APPROVED** [review]
Great job here, thanks for getting the delete vote mutation finished!

**Review by ETBassist — APPROVED** [review]
Wayne's happy

File: app/graphql/types/verse_type.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module Types
+  class VerseType < Types::BaseObject
+    field :id, ID, null: false
+    field :audio_path, String, null: false
+    field :user_id, Integer, null: false
+    field :title, String, null: false
+    field :competition_id, Integer, null: false
+    field :vote, Types::Vote, null: true
```
Comment (ETBassist) [inline app/graphql/types/verse_type.rb:8]: Here's hoping we don't run into too many conflicts here 😅

**Review by nmking22 — APPROVED** [review]
Great work, Sage!

**Pull request merged**